src/units.cpp

```diff
@@ -456,6 +457,32 @@ UnitsMap createUnitsMap(const UnitsPtr &units)
 {
     UnitsMap unitsMap;
     updateUnitsMap(units, unitsMap);
+
+    // dimensionality checks for exponents equal to zero in the map
+    auto it = unitsMap.begin();
+
+    while (it != unitsMap.end()) {
+        std::string unit = it->first;
+        if (it->second == 0.0) {
+            auto found = unitsMap.find("dimensionless");
+            if (found == unitsMap.end()) {
+                unitsMap.emplace(std::make_pair("dimensionless", 0.0));
+                it = unitsMap.erase(it);
+                --it;
+            } else {
+                it = unitsMap.erase(it);
+                --it;
+            }
```

Comment:
I don't understand your thinking here what is it that you are trying to achieve?

Reply:
The problem this part of the code was trying to address was issues when things become `dimensionless`. One problem I found when doing validation was that if we had units which reduced from some complicated expression to "meters^0.0, amperes^1.0, kg^2.0" and another unit which reduced to "seconds^0.0, amperes^1.0, kg^2.0", both those units should return `true` when passed into `equivalent()`. However, the `map` in its original form would keep `meters` and `seconds` in it, so the idea here is to add a `dimensionless` placeholder if anything has an `exponent` of 0.0, to eliminate such cases of unit cancellation returning the incorrect output.

Reply:
Although thinking about it now it's probably not necessary to do this; would be much cleaner to instead get rid of both any units with an `exponent` of 0.0, and any units with `dimensionless` in them.